Assets/MixedRealityToolkit/_Core/Resources/Shaders/MixedRealityStandard.shader

```diff
@@ -501,12 +527,33 @@ Shader "Mixed Reality Toolkit/Standard"
 #if defined(_INSTANCED_COLOR)
                 UNITY_SETUP_INSTANCE_ID(i);
 #endif
-                // Texturing.
+
+#if defined(_TRIPLANAR_MAPPING)
+                // Calculate triplanar uvs and apply texture scale and offset values like TRANSFORM_TEX.
+                fixed3 triplanarBlend = pow(abs(i.triplanarNormal), _TriplanarMappingBlendSharpness);
+                triplanarBlend /= dot(triplanarBlend, fixed3(1.0, 1.0, 1.0));
+                float2 uvX = i.triplanarPosition.zy * _MainTex_ST.xy + _MainTex_ST.zw;
+                float2 uvY = i.triplanarPosition.xz * _MainTex_ST.xy + _MainTex_ST.zw;
+                float2 uvZ = i.triplanarPosition.xy * _MainTex_ST.xy + _MainTex_ST.zw;
+
+                float3 axisSign = i.triplanarNormal < 0 ? -1 : 1;
```

Comment:
cg has a `sign` function which should be a bit faster and more idiomatic than this ternary http://developer.download.nvidia.com/cg/sign.html

This will have the added advantage of setting the `uv.x` values (below) to `0` when that component of the normal is 0 - essentially increasing the chance of a cache hit on the texture lookup when the normals match the basis vectors.

I've not profiled either of these :)

Reply:
Thank you for pointing this out. My gut reaction was to use sign too but I don't ever want a "axisSign" of zero, so I wasn't sure of a better way to do this?

Reply:
I don't have a problem with a sign of 0 in this case - its impact on the tri planar blend is exactly nothing, so it makes sense to me for its sign to match that. Any other choice is equally arbitrary IMHO.

Especially if it does give perf benefits - add a comment if you're worried about confusion.